streams-tests/shared/src/test/scala/zio/stream/ZChannelSpec.scala

```diff
@@ -823,7 +823,99 @@ object ZChannelSpec extends ZIOBaseSpec {
           _        <- ZChannel.scoped(fiberRef.locallyScoped(false)).runDrain
           value    <- fiberRef.get
         } yield assertTrue(value)
-      }
+      },
+      suite("pipeTo")(
+        test("identity.pipeTo(channel) is ZChannel.Suspend(channel)") {
+          val channel = ZChannel.refailCause(Cause.empty)
+          val result  = ZChannel.identity.pipeTo(channel)
+
+          assertTrue(result match {
+            case ZChannel.Suspend(effect) => effect() == channel
+            case _                        => false
+          })
+        },
+        test("identity.pipeTo(channel) produces correct results") {
+          val channel = ZChannel.writeAll(1, 2, 3)
+          val result  = ZChannel.identity.pipeTo(channel)
+
+          result.runCollect.map { tuple =>
+            assertTrue(tuple._1 == Chunk(1, 2, 3))
+          }
+        },
+        test("identity.pipeTo(channel) produces failure") {
+          val channel = ZChannel.fail("error")
+          val result  = ZChannel.identity.pipeTo(channel)
+
+          result.runDrain.exit.map { exit =>
+            assertTrue(exit == Exit.fail("error"))
+          }
+        },
+        test("channel.pipeTo(identity) is not skipped") {
+          val channel = ZChannel.writeAll(1, 2, 3)
+          val result  = channel.pipeTo(ZChannel.identity)
+
+          assertTrue(result match {
+            case ZChannel.PipeTo(_, _) => true
+            case _                     => false
+          })
+        }
```

Comment:
Do we actually need this test? In theory we could also optimize this path.

Note that I'm not suggesting that we should; I tried optimizing that path and added a `println` command to see if any of our tests would trigger the print to the console (yeah I'm a bit primitive in my ways 😅) and that optimization wouldn't produce any benefits, but I'm still unsure about explicitly testing that we don't optimize this path